script.js

```diff
@@ -18,18 +17,41 @@ async function fetchTodo(apiUrl) {
 
 // Append data in DOM 
 function appendData(data) {
-    const container = document.querySelector('.posts');
+    const container = document.querySelector('.post-container');
     data.forEach((item) => {
         const newTitle = document.createElement('div');
         newTitle.className = 'post';
         newTitle.innerHTML = item.title;
-        newTitle.addEventListener('click', () => {
-            console.log(item);
-        });
+        newTitle.addEventListener('click', (e) => {
+
+        showClickedPost(e.target);
+        backButton();
```

Comment:
Why does this function call on every post-click?
Can we call it just once within the click handler, wdyt?

Reply:
fix this